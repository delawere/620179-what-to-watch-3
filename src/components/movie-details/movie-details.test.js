import React from 'react';
import renderer from 'react-test-renderer';
import MovieDetails from './movie-details';

const cardData = {
  name: `test`,
  img: `img/test.jpg`
};

it(`MovieDetails renders correctly`, () => {
  const tree = renderer
    .create(<MovieDetails cardData={cardData}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
