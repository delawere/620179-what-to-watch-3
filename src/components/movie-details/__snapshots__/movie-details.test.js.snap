// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="test1"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="main.html"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="sign-in__link"
            href="/login"
            onClick={[Function]}
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            test1
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              test1Genre
            </span>
            <span
              className="movie-card__year"
            />
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                fill="#d9cd8d"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 5v14l11-7z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="test1 poster"
            height="327"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="/testURL"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="/testURL/details"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="/testURL/reviews"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div />
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="test1"
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              test1
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="test2"
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              test2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="test3"
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              test3
            </a>
          </h3>
        </article>
      </div>
    </section>
  </div>,
]
`;
